java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java

```diff
@@ -161,18 +161,27 @@ public static boolean isUnitTestRelatedToObjectMethods(@Nullable MethodTree meth
   }
 
   public static boolean isUnitTest(MethodTree methodTree) {
-    if (isOrOverridesJunit4TestMethod(methodTree)) {
+    boolean isSpringSanityTest = isSpringBootSanityTest(methodTree);
+    if (isOrOverridesJunit4TestMethod(methodTree) && !isSpringSanityTest) {
```

Comment:
This also disables all other rules that are applied to unit tests (e.g. S5961) for the `contextLoads` method. I think we might not want this and keep it specific for S2699 for now. Maybe it's better to move the `isSpringBootSanityTest` method into the implementation of S2699 and remove it from the test utils?